core/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -739,6 +746,13 @@ BigArrays createBigArrays(Settings settings, CircuitBreakerService circuitBreake
     }
 
     /**
+     * Build the {@link SearchModule}. Overridden by tests.
+     */
+    protected SearchModule createSearchModule(List<SearchPlugin> plugins) {
```

Comment:
But why does SearchModule need to create the SearchService? I find keeping all these "modules" around post guice just keeps things confusing (hard to know where something belongs) and am leaning towards everything being in Node (yes it would mean Node will be big, but it really is, right now there are arbitrary and inconsistent lines drawn).  You could move the single line binding SearchService out of SearchModule and add it to the other bindings done directly in Node.